### DISCUSSION TITLE
PIP-12: Automated Version Updates and Announcements in GUI and Daemon Client

### DISCUSSION THREAD

**Discussion by ja7ad** [pactus-project/PIPs#51] [PIPs]
PIP-12: [Automated Version Updates and Announcements in GUI and Daemon Client](https://pips.pactus.org/PIPs/pip-12)

**Comment by b00f** [comment]
@Ja7ad 
Your ideas are interesting and well-written. However, there are a few questions that might be good to discuss:

1. You mentioned, "For the GUI, a scheduled task will periodically check for updates." Why not check the version and available update on startup, similar to the CLI approach?
2. For the CLI, you suggest just logging the available update, correct?
3. The section under "Local JSON File Integration" seems unrelated to this PIP and could be moved to another PIP. This is due to a conflict with https://github.com/pactus-project/pactus/issues/823.

**Reply by ja7ad** [reply]
1. In the GUI, we are not limited to navigating a single path and process as in the CLI. We can perform n tasks in the background without requiring any specific allocation.we can systematically check for a new version after executing the GUI at specific intervals of n minutes. If a new version is available, we can notify the user through a pop-up announcements.

2. Before initializing the node in the CLI, we check for a new version from GitHub releases or other sources. Subsequently, we display a message asking whether to proceed with the update (y/n), this is onetime check:

```markdown
title
version
changes
-
-
-

are you want update to new version (y/n)?
```

3. Created new pip for this: https://github.com/pactus-project/PIPs/pull/52

**Comment by b00f** [comment]
We can't have an auto-update feature in Pactus. However, users may consider using a watcher application that:
- Checks for updates at regular intervals
- Shuts down their nodes
- Downloads the latest version
- Runs the node
